core/src/Kokkos_View.hpp

```diff
@@ -19,13 +19,1023 @@
+               std::is_same_v<typename base_t::layout_type, layout_left>) ||
+              (std::is_same_v<Layout, LayoutLeft> &&
+               std::is_same_v<typename base_t::layout_type,
+                              Experimental::layout_left_padded<> >) ||
+              (std::is_same_v<Layout, LayoutRight> &&
+               std::is_same_v<typename base_t::layout_type, layout_right>) ||
+              (std::is_same_v<Layout, LayoutRight> &&
+               std::is_same_v<typename base_t::layout_type,
+                              Experimental::layout_right_padded<> >),
+          void*> = nullptr)
+      : base_t(
+            handle,
+            Impl::mapping_from_array_layout<typename mdspan_type::mapping_type>(
+                arg_layout)) {}
+#else
+  // Constructors from legacy layouts when using Views of the new layouts
+  // LayoutLeft -> layout_left, layout_left_padded
+  // LayoutRight -> layout_right, layout_right_padded
+  // LayoutStride -> layout_stride
+  KOKKOS_FUNCTION
+  explicit inline View(const base_t::data_handle_type& handle,
+                       const LayoutStride& arg_layout)
+    requires(std::is_same_v<typename base_t::layout_type, layout_stride>)
+      : base_t(
+            handle,
+            Impl::mapping_from_array_layout<typename mdspan_type::mapping_type>(
+                arg_layout)) {}
+
+  KOKKOS_FUNCTION
+  explicit inline View(const base_t::data_handle_type& handle,
+                       const LayoutLeft& arg_layout)
+    requires(std::is_same_v<typename base_t::layout_type,
+                            Experimental::layout_left_padded<> >)
+      : base_t(
+            handle,
+            Impl::mapping_from_array_layout<typename mdspan_type::mapping_type>(
+                arg_layout)) {}
+
+  KOKKOS_FUNCTION
+  explicit inline View(const base_t::data_handle_type& handle,
+                       const LayoutRight& arg_layout)
+    requires(std::is_same_v<typename base_t::layout_type,
+                            Experimental::layout_right_padded<> >)
+      : base_t(
+            handle,
+            Impl::mapping_from_array_layout<typename mdspan_type::mapping_type>(
+                arg_layout)) {}
+
+  KOKKOS_FUNCTION
+  explicit inline View(const base_t::data_handle_type& handle,
+                       const LayoutLeft& arg_layout)
+    requires(std::is_same_v<typename base_t::layout_type, layout_left>)
+      : base_t(
+            handle,
+            Impl::mapping_from_array_layout<typename mdspan_type::mapping_type>(
+                arg_layout)) {}
+
+  KOKKOS_FUNCTION
+  explicit inline View(const base_t::data_handle_type& handle,
+                       const LayoutRight& arg_layout)
+    requires(std::is_same_v<typename base_t::layout_type, layout_right>)
+      : base_t(
+            handle,
+            Impl::mapping_from_array_layout<typename mdspan_type::mapping_type>(
+                arg_layout)) {}
+#endif
+
+#ifndef KOKKOS_ENABLE_CXX17
```

Comment:
The block right above this is `ifdef KOKKOS_ENABLE_CXX17 ... / else ...`, but this one's now starting with `ifndef KOKKOS_ENABLE_CXX17 ...`.
Can these blocks be consolidated or at least be converted to have the same ifdef conditions?
Just for some minorly better readability.

Reply:
They are different sets of ctors so I like to group them together separately (the C++17/20 versions)